pallets/connectors-gateway/src/lib.rs

```diff
@@ -0,0 +1,273 @@
+			<Self as frame_system::Config>::RuntimeOrigin,
+			Success = DomainAddress,
+		>;
+
+		/// The AdminOrigin ensures that some calls can only be performed by
+		/// admins.
+		type AdminOrigin: EnsureOrigin<<Self as frame_system::Config>::RuntimeOrigin>;
+
+		/// The incoming and outgoing message type.
+		///
+		/// NOTE - this `Codec` trait is the Centrifuge trait for connectors
+		/// messages.
+		type Message: Codec;
+
+		/// The message router type that is stored for each domain.
+		type Router: DomainRouter<Sender = Self::AccountId, Message = Self::Message>
+			+ Clone
+			+ Debug
+			+ MaxEncodedLen
+			+ TypeInfo
+			+ FullCodec
+			+ EncodeLike
+			+ PartialEq;
+
+		/// The type that processes incoming messages.
+		type InboundQueue: InboundQueue<Sender = DomainAddress, Message = Self::Message>;
+
+		type WeightInfo: WeightInfo;
+
+		/// Maximum size of an incoming message.
+		#[pallet::constant]
+		type MaxIncomingMessageSize: Get<u32>;
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub (super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		/// The router for a given domain was set.
+		DomainRouterSet { domain: Domain, router: T::Router },
+
+		/// A connector was added to a domain.
+		ConnectorAdded { connector: DomainAddress },
+
+		/// A connector was removed from a domain.
+		ConnectorRemoved { connector: DomainAddress },
+	}
+
+	/// Storage for domain routers.
+	///
+	/// This can only be set by an admin.
+	#[pallet::storage]
+	pub(crate) type DomainRouters<T: Config> = StorageMap<_, Blake2_128Concat, Domain, T::Router>;
+
+	/// Storage that contains a limited number of whitelisted connectors for a
+	/// particular domain.
+	///
+	/// This can only be modified by an admin.
+	#[pallet::storage]
+	pub(crate) type ConnectorsAllowlist<T: Config> = StorageDoubleMap<
+		_,
+		Blake2_128Concat,
+		Domain,
+		Blake2_128Concat,
+		DomainAddress,
+		(),
+		ValueQuery,
+	>;
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// Router initialization failed.
+		RouterInitFailed,
+
+		/// The origin of the message to be processed is invalid.
+		InvalidMessageOrigin,
+
+		/// The domain is not supported.
+		DomainNotSupported,
+
+		/// Message decoding error.
+		MessageDecodingFailed,
+
+		/// Connector was already added to the domain.
+		ConnectorAlreadyAdded,
+
+		/// Maximum number of connectors for a domain was reached.
+		MaxConnectorsReached,
+
+		/// Connector was not found.
+		ConnectorNotFound,
+
+		/// Unknown connector.
+		UnknownConnector,
+
+		/// Router not found.
+		RouterNotFound,
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		/// Set a domain's router,
+		#[pallet::weight(T::WeightInfo::set_domain_router())]
+		#[pallet::call_index(0)]
+		pub fn set_domain_router(
+			origin: OriginFor<T>,
+			domain: Domain,
+			router: T::Router,
+		) -> DispatchResult {
+			T::AdminOrigin::ensure_origin(origin)?;
+
+			ensure!(domain != Domain::Centrifuge, Error::<T>::DomainNotSupported);
```

Comment:
I guess we will remove this check as soon as we wrap up subsequent PRs right?

Reply:
This will stay, we can only remove this if we do some refactoring that gives us an associated type that we know is not a Centrifuge domain.

Reply:
We could have 

```rust
type Domain: Member + Parameter;

type LocalDomain: Get<Domain>;
```